### PR TITLE
Type assertion on nil value will fail.

### LINKED ISSUE #3996: Promote is failing
### Summary 
When an app builds and get promoted to staging(created manually, not during install), the build dies with an error. Apparently the error seems to be coming during replacing vault values in yaml.

### Steps to reproduce the behavior
1. jx install --no-default-environments=true
2. jx create env `staging, jx-staging`
3. jx import

### Expected behavior
App should be released in staging

### Actual behavior
Promotion dies with this error:

```
panic: interface conversion: interface {} is nil, not []interface {}
goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/vault.(*client).List(0xc00016ca20, 0x3f3e9a4, 0x7, 0x4705940, 0xc00016ca20, 0x0, 0x0, 0x101)
    /workspace/source/pkg/vault/vault_client.go:140 +0x35f
github.com/jenkins-x/jx/pkg/jx/cmd.(*StepHelmApplyOptions).fetchSecretFilesFromVault(0xc00021ca10, 0x3f383b5, 0x1, 0x46f1400, 0x8383798, 0xc000a411e0, 0x0, 0x1, 0x0, 0x0)
    /workspace/source/pkg/jx/cmd/step_helm_apply.go:284 +0x171
github.com/jenkins-x/jx/pkg/jx/cmd.(*StepHelmApplyOptions).Run(0xc00021ca10, 0x0, 0x0)
    /workspace/source/pkg/jx/cmd/step_helm_apply.go:176 +0x8ac
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdStepHelmApply.func1(0xc0006e0280, 0x8383798, 0x0, 0x0)
    /workspace/source/pkg/jx/cmd/step_helm_apply.go:68 +0x6f
github.com/spf13/cobra.(*Command).execute(0xc0006e0280, 0x8383798, 0x0, 0x0, 0xc0006e0280, 0x8383798)
    /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0xc00018aa00, 0xc0005f4a00, 0xc000884280, 0xc000645180)
    /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc00018aa00, 0xc000381090, 0x46d4380)
    /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc0008c9f88, 0xc0000c0058)
    /workspace/source/cmd/jx/app/jx.go:16 +0xed
main.main()
    /workspace/source/cmd/jx/jx.go:11 +0x22
```

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.128
Kubernetes cluster v1.12.7-gke.10
kubectl            v1.14.1
helm client        Client: v2.14.0+g05811b8
git                git version 2.20.1 (Apple Git-117)
Operating System   Mac OS X 10.14.4 build 18E226
```

### Jenkins type

- [x] Next Generation (Tekton + Prow)
- [ ] Classic Jenkins
- [ ] Serverless Jenkins (JenkinsFileRunner + Prow)

### Kubernetes cluster

GKE

### Operating system / Environment

Mac OS X

### PR DESCRIPTION
**Pull request by heroic**
Return early if secrets are nil to prevent panic when doing type assertion for looping.

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description
We are doing type assertion without verifying if the value is null or not. This can cause panic when we come across null

#### Which issue this PR fixes

fixes #3996

### CODE REVIEW

File: pkg/vault/vault_client.go [hunk 1]
```diff
@@ -145,10 +145,17 @@ func (v *client) List(path string) ([]string, error) {
 	if secrets == nil {
 		return secretNames, nil
 	}
+	
 	data := secrets.Data
 	if data == nil {
 		return secretNames, nil
 	}
+  
+  // Don't do type assertion on nil
```
Comment (hferentschik) [inline pkg/vault/vault_client.go]: I think the formatting is off here. You can run `make fmt` to check and fix formatting

File: pkg/vault/vault_client.go [hunk 2]
```diff
@@ -145,10 +145,17 @@ func (v *client) List(path string) ([]string, error) {
...
+	if secrets.Data["keys"] == nil {
```
Comment (ccojocar) [inline pkg/vault/vault_client.go]: I would refactor a bit this like this:

```Go
keys := secrets.Data["keys"]
if keys == nil {
    return secretNames, nil
}
for _, s := range keys.([]interface{}) {
...
}
```

**Pull request merged**